lib/dbservice_web/controllers/student_controller.ex

```diff
@@ -158,4 +163,72 @@ defmodule DbserviceWeb.StudentController do
       |> render("show.json", student: student)
     end
   end
+
+  def dropout(conn, %{"student_id" => student_id}) do
+    student = Users.get_student_by_student_id(student_id)
+
+    # Check if the student's status is already 'dropout'
+    if student.status == "dropout" do
+      conn
+      |> put_status(:bad_request)
+      |> json(%{error: "Student is already marked as dropout"})
+    else
+      user_id = student.user_id
+      current_time = DateTime.utc_now()
+
+      # Fetch status and group details in a single query
```

Comment:
why set group type when we already know it is gonna be "status"

Reply:
We are fetching the `group` here where `type` will is `status`. So basically not setting anything & just fetching. XD